src/main/java/io/javarig/RandomInstanceGenerator.java

```diff
@@ -25,35 +25,98 @@ private synchronized <T> T generate(Type type, Consumer<CollectionGenerator> col
         return generated;
     }
 
+    /**
+     * generate a random instance of the given type
+     *
+     * @param type the type of the object to generate
```

Comment:
@param type the type of the object to generate seems redundant, its clear for me that type is the type of the object to generate! If not a better way is to change the variable to objectType